src/Microsoft.VisualStudio.Sdk.TestFramework/MefHosting.cs

```diff
@@ -162,14 +171,35 @@ private async Task<IExportProviderFactory> CreateExportProviderFactoryAsync()
     /// <returns>A task whose result is the <see cref="ComposableCatalog"/>.</returns>
     private async Task<ComposableCatalog> CreateProductCatalogAsync()
     {
-        IEnumerable<Assembly> assemblies = this.catalogAssemblyNames.Select(Assembly.Load);
+        IEnumerable<Assembly> assemblies = this.catalogAssemblyNames.Select(this.LoadAssembly);
         DiscoveredParts discoveredParts = await PartDiscoverer.CreatePartsAsync(assemblies);
         ComposableCatalog catalog = ComposableCatalog.Create(Resolver.DefaultInstance)
             .AddParts(discoveredParts)
             .WithCompositionService();
         return catalog;
     }
 
+    private Assembly LoadAssembly(string name)
+    {
+        try
+        {
+            return Assembly.Load(name);
+        }
+        catch (FileNotFoundException)
+        {
+            // When using xUnit v3, some assemblies like `Mono.Cecil` and `xunit.abstractions`
+            // won't load using `Assembly.Load` even though the file exists in the AppDomain's
+            // base directory. For any assemblies that are not found, we'll try to load them
+            // by reading the file ourselves and loading them in memory.
+            if (DefaultAssemblyNamesToFileNames.TryGetValue(name, out string? fileName))
```

Comment:
Maybe we can fix this by removing Mono.Cecil and the other failing assemblies from the list. We can filter them out of the *.dll search.